frontend/public/components/utils/telemetry.ts

```diff
@@ -19,3 +20,21 @@ export const withoutSensitiveInformations = (location: Location): Location => {
     key: location.key,
   };
 };
+
+export const getTelemetryTitle = () => {
+  const titleElement = document.querySelector('title');
+  const titleValue = titleElement.getAttribute('data-telemetry');
+  if (!titleValue) {
+    const { productName } = getBrandingDetails();
+    const pageTitleValue = document.title;
+    const titleArray = pageTitleValue.split(' · ');
+    const last = titleArray[titleArray.length - 1];
+    if (titleArray.length === 1) {
+      return pageTitleValue;
+    } else if (titleArray.length && last === productName) {
+      const newTelemetrytitle = pageTitleValue.replace(` · ${productName}`, '');
+      return newTelemetrytitle;
+    }
+  }
+  return titleValue || document.title;
+};
```

Comment:
It works, but when I need to check the details and different paths here I want recommend a IMHO simpler solution. Wdyt about this:

```suggestion

const titleProductNameSuffix = ` · ${getBrandingDetails().productName}`;

export const getTelemetryTitle = () => {
  const titleElement = document.querySelector('title');
  let title = titleElement.getAttribute('data-telemetry') || document.title;
  if (title.endsWith(titleProductNameSuffix)) {
    title = title.substring(0, title.length - titleProductNameSuffix.length);
  }
  return title;
};
```